shpc/main/registry/__init__.py

```diff
@@ -171,18 +175,19 @@ def sync_from_remote(
 
         # No local registry provided, use default
         if not local:
-            local = Filesystem(self.settings.filesystem_registry)
+            local = self.filesystem_registry
 
-        tmpdir = remote.source
-        if tmpdir.startswith("http") or not os.path.exists(tmpdir):
-            tmpdir = remote.clone()
+        ini_remote = None
```

Comment:
What does ini_remote mean?

Reply:
`ini_remote` is the original registry, an instance of `VersionControl`, while `remote` is replaced with a local registry object (`Filesystem`) from the clone.

Reply:
ah so ini == initial? or init? I think I get it now - this is the step where we convert a remote registry to a filesystem with a clone, and you are storing an old version to check if the remote should be cleanup up, and (I think?) you only had to do this because the cleanup function was removed for the local filesystem? I think I understand now but maybe we can make it more clear for future developers. Given adding the filesystem cleanup back, maybe like:

```python
# Prepare to cleanup if we clone the remote to temporary location
cleanup = False
if isinstance(remote, VersionControl):
    remote = Filesystem(remote.clone())
    cleanup = True

...
if cleanup:
    remote.cleanup()
```
It really is the same logic you have, but using a named boolean that explicitly describes what is happening (ini_remote kind of threw me through some hoops trying to figure it out!) :laughing: